### DISCUSSION TITLE
PFsense + dvtws + Google

### DISCUSSION THREAD

**Discussion by vegetate7** [bol-van/zapret#256] [Q&A]
В общем вот такая связка. PFsense 2.7.2 и демон dvtws. Все работает, кроме некоторых гугловых сервисов. Похоже accounts.google.com и возможно другие. 
При этом перестают работать google.com и youtube, и прочее гугловое, если открыть залогиненую сессию.
Даже если вносить google.com в hostlist-exclude. И даже если оставить только hostlist с несколькими хостами.
в foreground режиме показывает, что ничего не тамперит:
```
hostname: www.google.com
Checking exclude hostlist
Hostlist check for www.google.com : negative
Hostlist check for google.com : positive
not applying tampering to this request
```
Но, видимо что-то такое с пакетами происходит, отчего гугл ломается :(
заруливаю пакеты так:
`ipfw delete 100 ; ipfw add 100 divert 989 tcp from not 192.168.0.0/16 to any 80,443 out not diverted not sockarg xmit pppoe0`  (пробовал и from any сначала, тоже не работало)
Пускаю так:
`/usr/local/sbin/dvtws --port 989 --dpi-desync=fake,split2 --dpi-desync-ttl=3`  плюс разные комбинации хостлистов.
Режимы подобрал блокчекером.

куда копать?

**Comment by bol-van** [comment]
ттл ломает близкие ggc
добавить --dpi-desync-fooling=md5sig

**Reply by vegetate7** [reply]
> ттл ломает близкие ggc добавить --dpi-desync-fooling=md5sig

fooling пробовал разные варианты. Собственно для этого и сделал пакеты только с внешнего адрес дивертить в демона, чтобы после НАТ попадали. И ttl пробовал 2. Разве что 1 не пробовал. Вечером попробую, когда все домашние от интернета отлипнут.
Но вопрос то немного не в том. Получается пакеты все равно изменяются, даже если хост не попадает в списки обхода?
Это только правилами ipfw можно будет побороть? Для гугла это дофига будет :(

**Reply by bol-van** [reply]
если в дебаге pass unmodified, значит не изменяются
можно проверить без параметров desync

**Reply by vegetate7** [reply]
> если в дебаге pass unmodified, значит не изменяются можно проверить без параметров desync

без параметров desync я так понимаю оно вообще ничего не должно менять? Попробовал. Все пакеты пишет 
`packet: id=1018 reinject unmodified`
но гугл все равно перестает работать. 
Еще интересная особенность. убрал из правила ipfw `xmit pppoe0`. И в логе стали видны пакеты идущие обратно. Типа 
```
packet: id=1007 len=40
IP4: 77.88.32.67 => 192.168.12.115 proto=tcp ttl=56 sport=40926 dport=80 flags=R seq=3516477272 ack_seq=0
packet: id=1007 reinject unmodified
```
что немного странно выглядит, dport вдруг стал портом хоста снаружи, а sport - моим портом, хотя src и dst остались как положено. Ну может особенности ipfw. Я больше по линуксам.
Ну в общем вот этих обратных пакетов нет в IP6. Может в этом проблема? Гугл вполне может оказаться на IP6

**Reply by bol-van** [reply]
есть подозрение, что pf иi ipfw вместе глючат

**Reply by vegetate7** [reply]
Понятно. Придется пробовать TPWS.

**Reply by sadok** [reply]
> есть подозрение, что pf иi ipfw вместе глючат

они не глючат. на исходящие первым отработает ipfw, потом pf. и наоборот

**Reply by arabesc** [reply]
> _есть подозрение, что pf иi ipfw вместе глючат_

у меня работают и проблем с google нет, но у меня ванильная фря 13.3, не PFsense
правда, автор упомняул  IPv6, у меня только IPv4
